### PR TITLE
islam_osman-assignment-w4-done

### PR DESCRIPTION
**Pull request by IslamAltayeb**

### CODE REVIEW

**Comment by IslamAltayeb** [comment]
actually, I don't know why it's committing all the branches every time, i tried many ways to commit the specific week but it's also committing all

File: Week4/homework/ex1-aggregation/aggregation.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            client.close();
+          }
+        });
+    })
+    .catch((err) => {
+      console.error('Error connecting to the database:', err);
+    });
+}
+
+importCSVData();
```
Comment (jrnhofman) [inline Week4/homework/ex1-aggregation/aggregation.js:59]: it is better not to have one big js file with all logic, can you split it?

File: Week4/homework/ex1-aggregation/aggregation.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+  return client.connect()
+    .then(() => {
+      console.log('db is connected');
+
+      const db = client.db(dbName);
+      const collection = db.collection(collectionName);
+
+      return collection.aggregate([
+        { $match: { Year: year, Age: age } },
```
Comment (jrnhofman) [inline Week4/homework/ex1-aggregation/aggregation.js:123]: the continent filter is missing here, this is returning results for all countries and continents!

File: Week4/homework/ex1-aggregation/aggregation.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        {
+          $group: {
```
Comment (jrnhofman) [inline Week4/homework/ex1-aggregation/aggregation.js:125]: the group is not incorrect but it is not needed, per year and age group you already have only one line per country, can you think of how to return everything plus an extra field here? Please check if your results match with the example in the exercise!

File: Week4/homework/ex2-transactions/transfer.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+const { MongoClient } = require('mongodb');
```
Comment (jrnhofman) [inline Week4/homework/ex2-transactions/transfer.js:1]: good job!

File: Week4/homework/ex2-transactions/setup.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  try {
+    await client.connect();
+    const db = client.db(dbName);
+
+    // Clean up the accounts collection
+    await db.collection('accounts').deleteMany({});
+
+    // Fill the accounts collection with sample data
+    const sampleData = [
+      { account_number: 101, balance: 1000, account_changes: [] },
```
Comment (jrnhofman) [inline Week4/homework/ex2-transactions/setup.js:19]: you took the 'easy' case here where account_changes is empty. Now let's assume there were already some previous account changes (you can make some up and add them here), does your code in transfer.js still work? (spoiler; it doesn't, the change number for example would not increment correctly).

Can you expand your sample data with some already existing account_changes and try to run your code again?

**Pull request closed without merging**